### ISSUE TITLE
Improve the orientation and navigation for screen reader users by using more headings and aria landmark roles

### ISSUE THREAD

**Issue by rpkoller** [bpmn-io/bpmn-js#1804]
__Is your feature request related to a problem? Please describe.__
I have to preface that I've mainly tested bpmn with the integration in ECA in Drupal but at the same time when writing up the issue created a free account in Camunda for testing purposes. I just needed points of reference integrating all aspects of bpmn like the properties panel, the element template chooser and so on.

If you are new to bpmn and get into a model for the first time and tab through the interface with the keyboard in ECA in Drupal the only elements i am able to reach are the bpmn logo link in the bottom right corner of the canvas, the properties panel with its section toggles as well as the form elements inside each section, and the button to save or cancel/leave a model. the toolbar is unreachable and invisible (in case i am a screen reader user), same goes for every model component on the canvas. Similar in Camunda, the toolbars aside the first visible focusable elements are the toggles in the properties panel as well as the contained form elements in case a section is expanded. similar to ECA in Drupal the toolbar and the model components aren't accessible at all by the keyboard. (on a side note in Camunda many focus states have either a way too low color contrast or aren't visible at all - WCAG 2.1 SC 2.4.7).

But screenreader users aren't navigating just by the tab key. When coming to a new page they mainly orient by the headings and landmark sections or in case they are familiar with the page use those for directly jumping in-between the different sections. See https://heydonworks.com/article/responses-to-the-screen-reader-strategy-survey/ (first question and it's answer), or https://youtu.be/OUDV1gqs9GA?t=402, and for a general overview see https://redish.net/wp-content/uploads/Theorfanos_Redish_InteractionsPaperAuthorsVer.pdf.
In Jaws for example you have the shortcut `h` for jumping in-between headings and another shortcut `r` for jumping in-between landmark roles (see the youtube link). In NVDA and Voiceover you also have the rotor menu with dedicated sections for headings and landmarks. I have only Voiceover available therefore i am unsure if NVDA and other screen readers also have shortcuts for headings and landmarks available, in Voiceover there is only one for headings. And i am unsure if on the other hand Jaws has a rotor menu like feature (i can't find one in the jaws cheatsheet at Deque: https://media.dequeuniversity.com/en/courses/generic/testing-screen-readers/2.0/docs/jaws-guide.pdf).

I have taken now a look at two solutions using bpmn (camunda and the eca module in drupal) 

**Camunda headings**
![Screen Shot 2023-01-04 at 16 53 38](https://user-images.githubusercontent.com/1665422/212508896-0cf0e6d3-e515-4977-b7b1-6829a112ccb4.jpg)
the only available heading for the entire page is the h3 as the title of the property panel - an h1 is missing as well as an h2 and the only heading the h3 is inside an aside element.

**Camunda landmarks**
![Screen Shot 2023-01-04 at 15 46 59](https://user-images.githubusercontent.com/1665422/212508953-c3e2f1e8-e38c-4d9f-8eac-d8464bcfbf24.jpg)

the only two landmarks relevant to bpmn are the main landmark wrapping everything bpmn related and the complementary landmark which is the properties panel. problem is main is unspecific and complementary is an abstract and unspecific term as well. (on a side note the complementary landmark aka the aside element is a top level items but in this case that top level item is wrapped inside the other top level item the main landmark. it isn't advised see https://dequeuniversity.com/rules/axe/4.2/landmark-complementary-is-top-level) 

**ECA headings** 
![Screen Shot 2023-01-04 at 15 44 23](https://user-images.githubusercontent.com/1665422/212508748-a5edbfd3-e8c8-408f-b2f0-f3ee92576ce0.jpg)
in ECA the only heading is Drupal specific it is the page title `condition new ECA model` in that case. it looks the h3 in the properties panel is Camunda specific. 

**ECA landmarks**
![Screen Shot 2023-01-04 at 15 43 57](https://user-images.githubusercontent.com/1665422/212508805-747ad133-f7dd-4f0d-9c92-59dcd2d19106.jpg)
in ECA the main landmark is the only bpmn specific landmark available wrapping everything. it looks the complementary landmark for the properties panels is also Camunda specific.  

__Describe the solution you'd like__
1) it might be good if sections of interactive elements on a model page are wrapped in landmark roles. maybe use a section element instead of a div (or add the `role="region"`) for the toolbar and the properties panel and perhaps also for the canvas with the actual model elements. And add also the `aria-label` attribute. the term that could be used for the aria-labels might be pretty straight forward. for example `toolbar`, `properties panel`, `model canvas` and maybe add an aria label to `main` element and rename it to something like `BPMN application`? 

2) since screenreader users not only orient by landmarks but also by headlines i wonder if it would make sense to also add a headline for each of the components. the properties panel already has one. but would it make sense to also add one to the toolbar and the model canvas? usually it is advisable to keep the content identical for screen reader users and sighted users but i wonder if it would be a reasonable step to make the headlines for the toolbar and the model canvas visually hidden. that way screenreader users would be able to grasp the structure of the page also by the headlines. for sighted users those would be a distraction and unnecessary in the context of the toolbar and the model canvas. 

_probably out of the scope of this issue but for context, making the page more structured by adding landmarks and headlines is first step but it would be still necessary to make the rest of the page also interactive and tab-able. for the toolbar there is already an open issue about adding shortcuts for the context menu and the toolbar  https://github.com/bpmn-io/bpmn-js/issues/768 and making each icon of the toolbar interactive that way. but it might be necessary to make the model components on the canvas interactive as well (same goes to enable opening the context menu for each component by the keyboard and not only by clicking with the mouse)_  

__Describe alternatives you've considered__

__Additional context__

**Comment by barmac** [comment]
Hi,

Thank you for your feedback and sharing your knowledge on the topic. If I understand it correctly, your proposal is to add regions to the Web Modeler which could look similar to:

![image](https://user-images.githubusercontent.com/28307541/213648043-82412fad-69cc-4bb4-a2ed-4c7099de6bb2.png)

Within the context of Web Modeler, the BPMN application (or BPMN diagram) part could be enclosed in HTML `main` tag.

I will forward this issue to the maintainers of Web Modeler, as this repository contains only the BPMN part.

**Comment by barmac** [comment]
As this is outside of the scope of the bpmn-js project, I am closing this issue. It's likely though that the Web Modeler is improved based on your feedback.

**Comment by nikku** [comment]
@barmac Interesting question here: Are palette, properties panel, and context-pad landmarks?

I think there is a couple of things that we can improve in the core:

> it might be good if sections of interactive elements on a model page are wrapped in landmark roles. maybe use a section element instead of a div (or add the role="region") for the toolbar and the properties panel and perhaps also for the canvas with the actual model elements.

**Comment by barmac** [comment]
My take on this is that the bpmn-js part of a website has role="application" as a whole: https://www.w3.org/TR/WCAG20-TECHS/ARIA11.html#ARIA11-description
